src/Symfony/Component/ClassLoader/ClassCollectionLoader.php

```diff
@@ -117,8 +117,17 @@ public static function load($classes, $cacheDir, $name, $autoReload, $adaptive =
 
         // cache the core classes
         if (!is_dir(dirname($cache))) {
-            mkdir(dirname($cache), 0777, true);
+            if (true !== @mkdir($cache, 0777, true)) {
+                $error = error_get_last();
+                if (!is_dir($cache)) {
+                    if ($error) {
+                        throw new \RuntimeException(sprintf('Failed to create "%s": %s.', $cache, $error['message']));
```

Comment:
All messages that I managed to get from mkdir are like this one: `mkdir(): Permission denied`
Which means the final message is going to be: `Failed to create "/foo/bar": mkdir(): Permission denied.`
I propose changing the message to: `preg_replace('/^(mkdir\(\): )?/', sprintf('mkdir(%s): ', $cache), $error['message'])`